### PR TITLE
Add sky k8s command runner

### PR DESCRIPTION
**Pull request by carolineechen**

### CODE REVIEW

**Comment by carolineechen** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1173?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1174** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1174?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1173** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1173?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1172** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1172?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1171** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1171?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1170** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1170?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @carolineechen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/hardware/sky/command_runner.py [hunk 1]
```diff
@@ -648,3 +648,193 @@ def rsync(
...
+
+        Example Usage:
+            runner = KubernetesCommandRunner((namespace, pod_name))
+            runner.run('ls -l')
+            runner.rsync(source, target, up=True)
+
+        Args:
+            node: The namespace and pod_name of the remote machine.
+        """
+        del kwargs
```
Comment (dongreenberg) [inline runhouse/resources/hardware/sky/command_runner.py:671]: I don't get this... but ok

**Review by dongreenberg — APPROVED** [review]
Sick work looks great!!!

**Review by dongreenberg — APPROVED** [review]

File: runhouse/resources/hardware/sky/command_runner.py (hunk 1 above)
Comment (carolineechen) [reply runhouse/resources/hardware/sky/command_runner.py:671]: this file is just source code from skypilot, that we subclass and edit for the runhouse versions. I guess the only core components for the k8s command runner are the namespace and pod name

**Pull request merged**